### PR TITLE
Torchinductor randn_like lowering

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #93005
* #93004
* #93003

Add lowering for randn_like, fixes https://github.com/pytorch/pytorch/issues/92368 by virtue of not taking a fallback path, although the 0-element prim stride is still incorrect. Would be nice to submit as a decomposition, but that is blocked by https://github.com/pytorch/pytorch/issues/92920. 

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -1116,6 +1119,20 @@ def randn(*args, **kwargs):
...
+    if config.fallback_random:
+        return fallback_randn_like(*args, **kwargs)
+    else:
+        dtype = kwargs.pop("dtype", None)
+        dtype = x.get_dtype() if dtype is None else dtype
+
+        device = kwargs.pop("device", None)
+        device = x.get_device() if device is None else device
+
+        return fast_randn(x.get_size(), dtype=dtype, device=device, **kwargs)
```
Comment (ngimel) [inline torch/_inductor/lowering.py:1133]: hm, this won't preserve permutation if `x` is permuted?

Comment (eellison) [reply torch/_inductor/lowering.py:1133]: it won't, but the MO thus far is we only set permutation requirements on graph outputs

**Review by ngimel — APPROVED** [review]
Can you add `rand_like` lowering while you are at it?

File: torch/_inductor/lowering.py [hunk 2]
```diff
@@ -1643,6 +1643,7 @@ def constant_like(fill_value):
 ones_like = create_tensor_like(tensor_constructor(1))
 if not config.fallback_random:
     rand_like = register_lowering(aten.rand_like)(create_tensor_like(rand))
+    randn_like = register_lowering(aten.randn_like)(create_tensor_like(randn))
```
Comment (ngimel) [inline torch/_inductor/lowering.py:1646]: lol nices, everything existed already

**Pull request closed without merging**